addon/components/docs-header/search-result/index.js

```diff
@@ -1,127 +1,114 @@
+    return Object.keys(metadata).reduce((matches, term) => {
+      let match = metadata[term];
+      let query = this.args.query;
+      let normalizedQuery = query.toLowerCase();
+      Object.keys(match).forEach((key) => {
+        if (key === 'text') {
+          let text = this.args.result.document.text;
+          let spaceIndices = text
+            .split('')
+            .map((char, index) => (char === ' ' ? index : null))
+            .filter((val) => val > 0);
 
-            match.text.position.forEach(([wordStart, length]) => {
-              let spaceAfterWord = spaceIndices.find((i) => i > wordStart);
-              let indexOfSpaceAfterWord = spaceIndices.indexOf(spaceAfterWord);
-              let indexOfSpaceBeforeWord = indexOfSpaceAfterWord - 1;
-              let indexOfStartingSpace =
-                indexOfSpaceBeforeWord > 3 ? indexOfSpaceBeforeWord - 3 : 0;
-              let indexOfEndingSpace =
-                indexOfSpaceAfterWord + 3 < spaceIndices.length
-                  ? indexOfSpaceAfterWord + 3
-                  : spaceIndices.length;
-              let matchingText = text.slice(
-                spaceIndices[indexOfStartingSpace],
-                spaceIndices[indexOfEndingSpace]
-              );
-              matchingText = this._highlight(
-                matchingText,
-                matchingText.indexOf(query),
-                query.length
-              );
+          match.text.position.forEach(([wordStart, length]) => {
+            let spaceAfterWord = spaceIndices.find((i) => i > wordStart);
+            let indexOfSpaceAfterWord = spaceIndices.indexOf(spaceAfterWord);
+            let indexOfSpaceBeforeWord = indexOfSpaceAfterWord - 1;
+            let indexOfStartingSpace =
+              indexOfSpaceBeforeWord > 3 ? indexOfSpaceBeforeWord - 3 : 0;
+            let indexOfEndingSpace =
+              indexOfSpaceAfterWord + 3 < spaceIndices.length
+                ? indexOfSpaceAfterWord + 3
+                : spaceIndices.length;
+            let matchingText = text.slice(
+              spaceIndices[indexOfStartingSpace],
+              spaceIndices[indexOfEndingSpace]
+            );
+            matchingText = this._highlight(
+              matchingText,
+              matchingText.indexOf(query),
+              query.length
+            );
 
-              matches.push(matchingText);
-            });
-          } else {
-            let normalizedTerm = term.toLowerCase();
-            this.get('result.document.keywords').forEach((keyword) => {
-              let normalizedKeyword = keyword.toLowerCase();
-              if (keyword.toLowerCase().indexOf(normalizedTerm) !== -1) {
-                let index = normalizedKeyword.indexOf(normalizedQuery);
-                matches.push(
-                  this._highlight(keyword, index, normalizedQuery.length)
-                );
-              }
-            });
-          }
-        });
+            matches.push(matchingText);
```

Comment:
There is a bug somewhere around this logic of finding matches for the searched text.
It is however not related to this update, since the problem is already present on ec-addon-docs.

Basically, it presents very weird matches for the query you type. I suggest we address this separately.